lib/ansible/modules/clustering/consul_acl.py

```diff
@@ -151,56 +158,48 @@ def execute(module):
 
 
 def update_acl(module):
-
     rules = module.params.get('rules')
-    state = module.params.get('state')
     token = module.params.get('token')
     token_type = module.params.get('token_type')
     mgmt = module.params.get('mgmt_token')
     name = module.params.get('name')
     consul = get_consul_api(module, mgmt)
     changed = False
 
-    try:
+    rules = decode_rules_as_yml(rules)
+    rules_as_hcl = encode_rules_as_hcl_string(rules) if len(rules) > 0 else None
 
+    try:
         if token:
             existing_rules = load_rules_for_token(module, consul, token)
-            supplied_rules = yml_to_rules(module, rules)
-            changed = not existing_rules == supplied_rules
+            changed = existing_rules != rules
```

Comment:
>  Looks like you're comparing HCL and JSON here.

I don't think that's true: it (should) be comparing two `RuleCollection` instances.


> ...as ansible detects a change each time I run. 

The tests reveal that this happens when setting a new token - there is no check for whether a token with the same name has been created before. I'll fix this, along with any other idempotence issues that lie after this one. After I fix the issues highlighted in the new tests, it would be great if you could verify the fixes with what you're running. I'll let you know when I'm done.